apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -512,6 +493,74 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
             return logEntry;
         }
 
+        private static Value BuildJsonValue(object objectValue)
+        {
+            var typeCode = Convert.GetTypeCode(objectValue);
+            switch (typeCode)
+            {
+                case TypeCode.Object:
+                    if (objectValue is System.Collections.IDictionary dictionary)
+                    {
+                        var dictionaryStruct = new Struct();
+                        var itemEnumerator = dictionary.GetEnumerator();
+                        while (itemEnumerator.MoveNext())
+                        {
+                            var dictionaryKey = itemEnumerator.Key?.ToString();
+                            if (string.IsNullOrEmpty(dictionaryKey))
+                            {
+                                continue;
+                            }
+
+                            var itemTypeCode = Convert.GetTypeCode(itemEnumerator.Value);
+                            dictionaryStruct.Fields.Add(dictionaryKey, BuildSimpleJsonValue(itemTypeCode, itemEnumerator.Value));
+                        }
+                        return Value.ForStruct(dictionaryStruct);
+                    }
+                    else if (objectValue is System.Collections.ICollection collection)

```

Comment:
Please make this support `IEnumerable`, not just `ICollection`. This will avoid confusion from people who don't know this code who may well assume it'll support `IEnumerable`.
And please use Linq rather than the `foreach` loop if possible.

Reply:
I use foreach as it allows a single array allocation, instead of first doing list-allocation (via Linq) and then toarray-allocation.

Reply:
OK, sounds reasonable, can you put a comment in saying that?

Reply:
Added comment